### ISSUE TITLE
Pre-trained models do not seem to restore correctly

### ISSUE THREAD

**Issue by jrc-exp** [TorchDSP/torchsig#8]
I was able to modify the code in a PR to get the pre-trained model weights to restore correctly, but when I run the pre-trained models against the validation dataset they do not produce meaningful results. 

I used the following script to generate the impaired Sig53 dataset, and restore the EfficientNet B4 weights. And I used the evaluation code from the example notebook to run predict with the model as follows. 

```python
from pathlib import Path

import numpy as np
import torch
import torchsig
import torchsig.transforms as ST

from matplotlib import pyplot as plt
from sklearn.metrics import classification_report
from tqdm import tqdm

from cm_plotter import plot_confusion_matrix
from torchsig.datasets import Sig53

root = 'sig53'

Path(root).mkdir(parents=True, exist_ok=True)
train = False
impaired = False
class_list = list(Sig53._idx_to_name_dict.values())
transform = ST.Compose([
    ST.RandomPhaseShift(phase_offset=(-1, 1)),
    ST.Normalize(norm=np.inf),
    ST.ComplexTo2D(),
])
target_transform = ST.DescToClassIndex(class_list=class_list)

sig53_impaired_val = Sig53(
    root=root, 
    train=False, 
    impaired=True,
    transform=transform,
    target_transform=target_transform,
    use_signal_data=True,
)

# Load the pretrained weights directly downloaded from torchsig.com
model = torchsig.models.efficientnet_b4(
    pretrained=True,
    path="efficientnet_b4_online.pt",
)

device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
model = model.to(device)
model.eval()

# Infer results over validation set
num_test_examples = len(sig53_impaired_val)
num_classes = len(list(Sig53._idx_to_name_dict.values()))
y_raw_preds = np.empty((num_test_examples,num_classes))
y_preds = np.zeros((num_test_examples,))
y_true = np.zeros((num_test_examples,))

for i in tqdm(list(range(0,num_test_examples))):
    # Retrieve data
    idx = i # Use index if evaluating over full dataset
    data, label = sig53_impaired_val[idx]
    # Infer
    data = torch.from_numpy(np.expand_dims(data,0)).float()
    data = data.cuda() if torch.cuda.is_available() else data
    with torch.no_grad():
        pred_tmp = model(data)
    pred_tmp = pred_tmp.cpu().numpy() if torch.cuda.is_available() else pred_tmp
    # Argmax
    y_preds[i] = np.argmax(pred_tmp)
    # Store label
    y_true[i] = label

acc = np.sum(np.asarray(y_preds)==np.asarray(y_true))/len(y_true)
plot_confusion_matrix(
    y_true, 
    y_preds, 
    classes=class_list,
    normalize=True,
    title="Example Modulations Confusion Matrix\nTotal Accuracy: {:.2f}%".format(acc*100),
    text=False,
    rotate_x_text=90,
    figsize=(16,9),
)

plt.savefig('confusion_matrix_effnetb4_online.png')
```

However, the final prediction is seemingly random:
![confusion_matrix_restored_effnetb4_online](https://user-images.githubusercontent.com/55510457/193288966-8b080e75-86c9-4975-807a-5ae1d312942e.png)

I confirmed that the weights are restoring into the parameters of the model when using the `pretrained=True` flag, and that you get randomly initialized weights when you run with `pretrained=False`. 

The major difference is that I'm using the code as posted to create the impaired dataset, and I guess there is potential that my randomized dataset does not match the dataset that you used. 

However, training the "clean" model for 1M steps using the example notebook on the clean dataset and evaluating it on the clean validation set did result in a model that at least learned something:
![confusion_matrix_trained_effnetb4_static](https://user-images.githubusercontent.com/55510457/193289551-ff306dd3-b38a-4eb6-8fbc-b2799b85a719.png)

Is there something in my above script that isn't correct? I've tested the same code restoring the XCiT-Tiny12 network with the same outcome.

**Comment by jrc-exp** [comment]
I had not re-installed torchsig after you pushed the update to pin the version of the timm module. When I use the pinned timm module version the model does restore as intended, and I get the expected result. 

![confusion_matrix_effnetb4_online](https://user-images.githubusercontent.com/55510457/193329546-4cdefed5-5a7f-49b8-b06e-d76b767082de.png)

Thanks!
